follow_redirects set to true is ignored if first redirect request ends with a too many redirects stack-trace.

To be more specific - if the test specifies max_retries with a delay_before or delay_after value then the following behaviour is observed:

```
INFO     tavern._plugins.rest.request:request.py:219 Overriding global follow_redirects setting of False with test-level specification of True
DEBUG    tavern._plugins.rest.request:request.py:223 Allow redirects in stage: True
DEBUG    tavern._plugins.rest.request:request.py:433 Request args: {'method': 'GET', 'url': '<REDACTED>', 'verify': True, 'stream': False, 'allow_redirects': True}
DEBUG    tavern.plugins:plugins.py:243 Getting expected response for requests
DEBUG    tavern.util.dict_util:dict_util.py:137 Not formatting something of type '<class 'int'>'
DEBUG    tavern.util.dict_util:dict_util.py:137 Not formatting something of type '<class 'float'>'
DEBUG    tavern.util.delay:delay.py:23 Delaying before request for 10.00 seconds
INFO     tavern.core:core.py:244 Running stage : serving an Admin Welcome login page
DEBUG    urllib3.connectionpool:connectionpool.py:943 Starting new HTTPS connection (1): <REDACTED>
DEBUG    urllib3.connectionpool:connectionpool.py:442 https://tt.anise.pzt.dev.perfsec.com:443 "GET <REDACTED> HTTP/1.1" 302 0
DEBUG    urllib3.connectionpool:connectionpool.py:442 https://tt.anise.pzt.dev.perfsec.com:443 "GET <REDACTED> HTTP/1.1" 302 0
DEBUG    urllib3.connectionpool:connectionpool.py:442 https://tt.anise.pzt.dev.perfsec.com:443 "GET <REDACTED> HTTP/1.1" 302 0
DEBUG    urllib3.connectionpool:connectionpool.py:442 https://tt.anise.pzt.dev.perfsec.com:443 "GET <REDACTED> HTTP/1.1" 302 0
DEBUG    urllib3.connectionpool:connectionpool.py:442 https://tt.anise.pzt.dev.perfsec.com:443 "GET <REDACTED> HTTP/1.1" 302 0
DEBUG    urllib3.connectionpool:connectionpool.py:442 https://tt.anise.pzt.dev.perfsec.com:443 "GET <REDACTED> HTTP/1.1" 302 0
...
ERROR    tavern._plugins.rest.request:request.py:472 Error running prepared request
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/tavern/_plugins/rest/request.py", line 470, in run
    return self._prepared()
  File "/usr/local/lib/python3.7/site-packages/tavern/_plugins/rest/request.py", line 455, in prepared_request
    return session.request(**self._request_args)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 665, in send
    history = [resp for resp in gen]
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 665, in <listcomp>
    history = [resp for resp in gen]
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 166, in resolve_redirects
    raise TooManyRedirects('Exceeded {} redirects.'.format(self.max_redirects), response=resp)
requests.exceptions.TooManyRedirects: Exceeded 30 redirects.
INFO     tavern.util.retry:retry.py:55 Stage 'serving an Admin Welcome login page' failed for 1 time. Retrying.
DEBUG    tavern.plugins:plugins.py:210 Initialising request class for requests (<class 'tavern._plugins.rest.request.RestRequest'>)
DEBUG    tavern.util.dict_util:dict_util.py:137 Not formatting something of type '<class 'NoneType'>'
DEBUG    tavern._plugins.rest.request:request.py:248 No cookies specified in request, sending all
DEBUG    tavern._plugins.rest.request:request.py:223 Allow redirects in stage: False
DEBUG    tavern._plugins.rest.request:request.py:433 Request args: {'method': 'GET', 'url': '<REDACTED>', 'verify': True, 'stream': False, 'allow_redirects': False}
DEBUG    tavern.plugins:plugins.py:243 Getting expected response for requests
DEBUG    tavern.util.dict_util:dict_util.py:137 Not formatting something of type '<class 'int'>'
DEBUG    tavern.util.dict_util:dict_util.py:137 Not formatting something of type '<class 'float'>'
DEBUG    tavern.util.delay:delay.py:23 Delaying before request for 10.00 seconds
INFO     tavern.core:core.py:244 Running stage : serving an Admin Welcome login page
DEBUG    urllib3.connectionpool:connectionpool.py:442 <REDACTED> "GET <REDACTED> HTTP/1.1" 302 0
DEBUG    tavern.plugins:plugins.py:273 Initialising verifier for requests (<class 'tavern._plugins.rest.response.RestResponse'>)
INFO     tavern._plugins.rest.response:response.py:50 Response: '<Response [302]>'
DEBUG    tavern._plugins.rest.response:response.py:64 Headers:
```
It is understood that redirect loops are very bad but the test is expected to behave as directed and if it can't then it should just fail the test with the first redirect stack-trace instead of retrying with the default value of false for follow_redirects.

Observed on version 1.4.0 and 1.16.1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -41,7 +41,7 @@ jobs:
           python: "3.9"
           TOXCFG: tox-integration.ini
         - TOXENV: pypy3-noextra
-          python: pypy3
+          python: pypy-3.7-v7.x
           TOXCFG: tox-integration.ini
         - TOXENV: py39-mqtt
           python: "3.9"

--- a/example/advanced/testing_utils.py
+++ b/example/advanced/testing_utils.py
@@ -11,6 +11,7 @@ def assert_quick_response(response):
     """
     assert response.elapsed < datetime.timedelta(seconds=0.1)
 
+
 def create_bearer_token():
     # Authorization: "bearer {test_login_token:s}"
 
@@ -25,6 +26,4 @@ def create_bearer_token():
 
     token = jwt.encode(payload, SECRET, algorithm="HS256").decode("utf8")
 
-    return {
-        "Authorization": "Bearer {}".format(token)
-    }
+    return {"Authorization": "Bearer {}".format(token)}

--- a/example/mqtt/testing_utils.py
+++ b/example/mqtt/testing_utils.py
@@ -1,4 +1,3 @@
 def message_says_hello(msg):
-    """Make sure that the response was friendly
-    """
+    """Make sure that the response was friendly"""
     assert msg.payload.get("message") == "hello world"

--- a/tavern/core.py
+++ b/tavern/core.py
@@ -1,4 +1,5 @@
 from contextlib import ExitStack
+import copy
 from copy import deepcopy
 from distutils.util import strtobool
 import functools
@@ -238,6 +239,7 @@ def run_stage(sessions, stage, test_block_config):
         stage (dict): specification of stage to be run
         test_block_config (dict): available variables for test
     """
+    stage = copy.deepcopy(stage)
     name = stage["name"]
 
     attach_stage_content(stage)

--- a/tests/integration/server.py
+++ b/tests/integration/server.py
@@ -267,6 +267,19 @@ def redirect_to_other_endpoint():
     return redirect("/redirect/destination", 302)
 
 
+@app.route("/redirect/loop", methods=["GET"])
+def redirect_loop():
+    try:
+        if redirect_loop.tries > 50:
+            return redirect("/redirect/destination", 302)
+        else:
+            redirect_loop.tries += 1
+    except AttributeError:
+        redirect_loop.tries = 1
+
+    return redirect("/redirect/loop", 302)
+
+
 @app.route("/redirect/destination", methods=["GET"])
 def get_redirected_to_here():
     return jsonify({"status": "successful redirect"}), 200

--- a/tests/integration/test_follow_redirects.tavern.yaml
+++ b/tests/integration/test_follow_redirects.tavern.yaml
@@ -11,6 +11,21 @@ stages:
       url: "{host}/redirect/source"
     response:
       status_code: 302
+---
+
+includes:
+  - !include common.yaml
+
+test_name: Test redirecting loops
+
+stages:
+  - name: Expect a 302 without setting the flag
+    max_retries: 2
+    request:
+      follow_redirects: true
+      url: "{host}/redirect/loop"
+    response:
+      status_code: 200
 
 ---
 

